internal/gatewayapi/translator.go

```diff
@@ -0,0 +1,515 @@
+package gatewayapi
+
+import (
+	"fmt"
+
+	"github.com/envoyproxy/gateway/pkg/ir"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	KindGateway   = "Gateway"
+	KindHTTPRoute = "HTTPRoute"
+	KindService   = "Service"
+)
+
+// Cache is an interface that the Translator requires
+// for getting Gateway API and related resources.
+type Cache interface {
```

Comment:
is there a benefit to making this an interface ?
maybe making Cache concrete with
```
type Cache struct {
  Gateways []*v1beta1.Gateway
  HTTPRoutes []*v1beta1.HTTPRoute
  Services []*v1.Service
}
```
and helper funcs for `Cache` can help test the message/API better since it will be strongly typed ?

also wdyt of `Resources` instead of `Cache` ?

Reply:
Yeah that's fine if that works for the Providers, I wasn't sure if they'd want to provide a struct or have method calls. I'm also good with the naming change.

Reply:
restructured and renamed in https://github.com/envoyproxy/gateway/pull/155/commits/cff94b3a4c8be011398f878ecc3bd830382b0067